### PR TITLE
docs(datasets) Add recommended fl datasets docs

### PR DESCRIPTION
**Pull request by adam-narozniak**
Add a docs page that:
* lists the recommended datasets 
* has division by modules (image, audio, tabular, text)
* has some module specific information

### CODE REVIEW

File: datasets/doc/source/recommended-fl-datasets.rst [hunk 1]
```diff
@@ -0,0 +1,163 @@
+Recommended FL Datasets
+=======================
+
+This page lists the recommended datasets for federated learning research, which can be used with Flower Datasets ``flwr-datasets``.
+
+.. note::
+
+    All datasets from HuggingFace Hub can be used with our library. This page presents just a set of datasets we collected that you might find useful.
```
Comment (jafermarq) [inline datasets/doc/source/recommended-fl-datasets.rst:8]:
```suggestion
    All datasets from `HuggingFace Hub <https://huggingface.co/datasets>`_ can be used with our library. This page presents just a set of datasets we collected that you might find useful.
```

Comment (jafermarq) [reply datasets/doc/source/recommended-fl-datasets.rst:8]: worth adding?

**Review by jafermarq — COMMENTED** [review]
Hey @adam-narozniak, this looks very good! 

Worth adding somewhere a link to the [fds-quickstart](https://flower.ai/docs/datasets/tutorial-quickstart.html) and potentially also some of the Flower quickstart examples (e.g. https://github.com/adap/flower/tree/main/examples/quickstart-pytorch) for a "complete" example ?

File: datasets/doc/source/recommended-fl-datasets.rst (hunk 1 above)
Comment (adam-narozniak) [reply datasets/doc/source/recommended-fl-datasets.rst]: Done

**Review by jafermarq — APPROVED** [review]

**Pull request merged**